internal/filter/filterconfig/config.go

```diff
@@ -75,7 +75,7 @@ type MatchProperties struct {
 	// For logs, one of LogNames, Attributes, Resources or Libraries must be specified with a
 	// non-empty value for a valid configuration.
 
-	// For metrics, one of MetricNames, Expressions, or ResourceAttributes must be specified with a
+	// For metrics, one of MetricNames or ResourceAttributes must be specified with a
```

Comment:
Are we removing Expressions support with this PR?

Reply:
It's hard for me to remember why I did this because I first committed it in December of last year, but I think the reason is because there isn't a field called Expressions, so saying you must specify it doesn't make much sense. Let me know if there is a reason it is mentioned.

Reply:
following that logic I probably should have also removed references to ResourceAttributes as well

Reply:
It looks like the matcher moved from `internal/filter/filtermetric` drops support of `match_type: expr` config option which is a significant breaking change not strictly related to this PR. I'm still not sure why can't we extent `internal/filter/filtermetric` to support metric attributes instead of moving the logic to this component

Reply:
do you want me to do that?

Reply:
I would prefer that, yes, if you still intend to have this PR merged. However, if it's no longer relevant to you, we can close it. This is because it will eventually be replaced with OTTL, which offers more capabilities. I apologize for the significant delay, this PR escaped my attention for a while.

Reply:
Okay, I think I’m going to close it because I have honestly lost context on this PR over the last year waiting for review. If someone else wants to take it over, they can base it off my fork.